### DISCUSSION TITLE
[@types/eslint] `ESLint.Formatter['format']` should take only one argument

### DISCUSSION THREAD

**Discussion by fasttime** [DefinitelyTyped/DefinitelyTyped#58761] [Issues with a @types package]
The [`Formatter`](https://eslint.org/docs/developer-guide/nodejs-api#-formatter-type) type has a `format` method that should accept an array of results as its only argument.

This reflects the way this method is called by ESLint: https://github.com/eslint/eslint/blob/v8.9.0/lib/cli.js#L181

By contrast, the current type definition includes a second argument which is inconsistent with both the documentation and the usage:

```ts
    interface Formatter {
        format(results: LintResult[], data?: LintResultData): string | Promise<string>;
    }
```

I would like to see this changed to:

```ts
    interface Formatter {
        format(results: LintResult[]): string | Promise<string>;
    }
```

[a comment by an automated account is not shown]

**Comment by bradzacher** [comment]
This is incorrect.
That line you linked is where ESLint calls its **internal** format method, which is created by `loadFormatter`, and then calls the formatter here:
https://github.com/eslint/eslint/blob/781f8d0e2400f4a3c4b633b43d318ef91767c6c4/lib/eslint/eslint.js#L650-L661

**Reply by fasttime** [reply]
I know that. I suppose that `Formatter` is this internal object returned by `loadFormatter`. At least it should be according to the [documentation](https://eslint.org/docs/developer-guide/nodejs-api#-formatter-type).

I don't think there is a typing for functions exported by eslint-plugin- formatters at this time (which would indeed accept a second parameter).

**Comment by fasttime** [comment]
To be clear: there is no typing for formatter functions (those exported by plugins), which take an additional second parameter. Only for wrapper objects created by `eslint.loadFormatter`. These objects have a method `format` that only accepts one parameter. Not only that, any additional parameter provided to the `format` method would be ignored, specifically because the second parameter passed to the underlying function is automatically added by the wrapping logic in `loadFormatter`.
